pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -545,23 +545,31 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
       // Decode message
       StreamDataDecoderResult decodedRow = _streamDataDecoder.decode(messagesAndOffsets.getStreamMessage(index));
       RowMetadata msgMetadata = messagesAndOffsets.getStreamMessage(index).getMetadata();
+      GenericRow decoderResult;
       if (decodedRow.getException() != null) {
-        // TODO: based on a config, decide whether the record should be silently dropped or stop further consumption on
-        // decode error
-        realtimeRowsDroppedMeter =
-            _serverMetrics.addMeteredTableValue(_metricKeyName, ServerMeter.INVALID_REALTIME_ROWS_DROPPED, 1,
-                realtimeRowsDroppedMeter);
+        if (_tableConfig.getIngestionConfig() != null
+            && _tableConfig.getIngestionConfig().isContinueOnError()) {
+          decoderResult = null;
+          realtimeRowsDroppedMeter =
+              _serverMetrics.addMeteredTableValue(_metricKeyName, ServerMeter.INVALID_REALTIME_ROWS_DROPPED, 1,
+                  realtimeRowsDroppedMeter);
+        } else {
+          throw new RuntimeException("Caught exception while decoding record", decodedRow.getException());
```

Comment:
Marked PR as `backward-incompat` because of this line. I think some change was done in 0.10 release because of which we were simply returning `null` here and not throwing exception. However, this PR addresses a `TODO` present in that code
```java
 // TODO: based on a config, decide whether the record should be silently dropped or stop further consumption on
        // decode error
 ```

Reply:
@navina Let me know what I should do here

Reply:
Ideally, we would want users to "opt-in" for `continueOnError`. So, I think it should be fine to make this incompatible change. The default behavior should be to stop the ingestion on error. 

@npawar any concerns with this backward incompatible change?